moped-editor/src/views/staff/StaffListViewCustomComponents.js

```diff
@@ -30,3 +40,118 @@ export const EditUserButton = ({id}) => (
     }
   />
 );
+
+/**
+ * Filter function to identify moped user group (MUG) members
+ * @param {object} user - a `moped_user` object
+ * @returns user
+ */
+const mugUserFilter = (user) => user.is_user_group_member && !user.is_deleted;
+
+/**
+ * Filter function to identify all active users who are not non-login users
+ * @param {object} user - a `moped_user` object
+ * @returns user
+ */
+const allUserFilter = (user) =>
+  !user.is_deleted && !user.roles.includes("non-login-user");
+
+/**
+ * Copies user email addresses to clipboard as a semi-colon-separated string
+ * @param {object[]} users - array of `moped_user` bjects
+ * @param {function} userFilter - filter function to apply to user array
+ * @returns Promise to write data to clipboard
+ */
+const copyEmailsToClipboard = (users, userFilter) => {
+  const emails = users
+    .filter(userFilter)
+    .map((u) => u.email)
+    .join(";");
+  const type = "text/plain";
+  const blob = new Blob([emails], { type });
+  const data = [new ClipboardItem({ [type]: blob })];
+  return navigator.clipboard.write(data);
+};
+
+/**
+ * Menu component which copies Moped user emails to clipboard
+ * @param {object[]} users - array of `moped_user` objects
+ */
+export const CopyMugUsersButton = ({ users }) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [copiedListName, setCopiedListName] = useState(null);
+
+  const handleMenuClick = (e) => {
+    setAnchorEl(e.currentTarget);
+  };
+
+  const handleCloseMenu = useCallback(() => {
+    setAnchorEl(null);
+  }, [setAnchorEl]);
+
+  const onMenuItemClick = (name) => {
+    const userFilter = name === "mug" ? mugUserFilter : allUserFilter;
+    copyEmailsToClipboard(users || [], userFilter).then(() => {
+      setCopiedListName(name);
+    });
+  };
+
+  useEffect(() => {
+    /**
+     * Effect which closes the menu after a brief pause
+     */
+    if (!copiedListName) return;
+    const timeout = setTimeout(() => {
+      setCopiedListName(null);
+      handleCloseMenu();
+    }, 300);
+    return () => clearTimeout(timeout);
+  }, [copiedListName, handleCloseMenu]);
+
+  const menuOpen = Boolean(anchorEl);
+
+  return (
+    <div>
```

Comment:
dunno if you want to give all users the ability to download the emails or only admins. i guess limiting who can see the button would just make the staff view "cleaner"